New graph screen

jujinfu opened the pull request.

panjiw committed:
Added folder for graph images

panjiw committed:
New table for graph screen

panjiw committed:
Graph screen php class

panjiw committed:
fixed file requirement

panjiw committed:
fixed table patching for gs table

panjiw committed:
fixed a missing tag

panjiw committed:
New graph options template and save

panjiw committed:
more option saving changes

panjiw committed:
lists screen graphs

panjiw committed:
new columns to the table

panjiw committed:
added new setting tab template, fixed some things

panjiw committed:
the library for color picking

panjiw committed:
more files for the new template

panjiw committed:
updated graph screen with new fields, and bunch of fixes

panjiw committed:
added name sanitizing function

panjiw committed:
fixed sql query

eikonos committed:
Fix quotes in table creation statement.

panjiw committed:
Changed graph image naming, added an example for the bug

panjiw committed:
Merge branch 'graph_screen' of https://github.com/jegelstaff/formulize into graph_screen

panjiw committed:
fixed label bug

panjiw committed:
a bunch of code from entries display, not working yet

panjiw committed:
Added the option to pick default view

Still very dirty but filtered data from selected view is now available (has been printed) not yet put into graphDisplay process though

panjiw committed:
removed all the print used for development

panjiw committed:
Fixed the menu link creation problem, fixed in another branch but easier to just do it again here

panjiw committed:
removed the blank default view option because it doesn't make sense as there are no other functions other than graphing

panjiw committed:
Cleaned up and commented, the most basic of functions now works

panjiw committed:
Removed prints used for debugging

panjiw committed:
Added the current view options

panjiw committed:
Fixed the filter, now actually works

panjiw committed:
Cleaned up the screen, dynamic view selection works but not as advanced as the one in entriesdisplay

jegelstaff committed:
Fixing incorrect syntax for arrays

Commit:
database patch, creating fixed, check graph params

eikonos commented on modules/formulize/class/graphScreen.php:
```diff
@@ -0,0 +1,128 @@
+			"usecurrentviewlist" => $screen->getVar('usecurrentviewlist')
+		);
+		include_once XOOPS_ROOT_PATH."/modules/formulize/include/graphdisplay.php";
+		displayGraph('Bar', $screen->getVar('fid'), $screen->getVar('frid'), $screen->getVar('labelelem'), $screen->getVar('dataelem'), $screen->getVar('ops'), $options);
```
There's a problem here. These two variables are integers, which represent element IDs: $screen->getVar('labelelem'), $screen->getVar('dataelem'). Later in graphdislay.php, these are used like this: display($entry, $dataElement);. It works if the element has the same ID and handle, but if the element handle has been changed to text, such as, 'client_name' then it fails. The code in one of these places should be changed so that it's consistently using element IDs or handles, and it's probably better to use handles since that's what a lot of the other code does.

jujinfu commented:
at the beginning, all graphs are stored with label element id and data element id. And later it should be using element id any where else. I didn't notice this before. should I change all to handles? that means change from very beginning. or i just fix this by using id in other places?

Commit:
change element identifier to handles

In graphdisplay.php, using element handle instand of element id. in
graphScreen.php, convert element id to element handle before calling
dispalyGraph

Commit:
Added Pie Chart

jujinfu commented:
so far i have changed a lot in graphdisplay.php. now we will process all data then according to the graph type to generate the graph. still need some small improvements

Commit:
minor fixes

Added a field to show what is the type of graph screen
A link can go back to screen lists

eikonos committed:
Merge branch 'master' into new-graph-screen

mmacke01 committed:
merge master into new-graph-screen

JamesEarle committed:
adjustments to screen graph cases and options. Fixed bug with default values

mmacke01 committed:
improving requests for input determining what to graph

mmacke01 committed:
removed chance of error making graphs due to unset label or data fields - set label field same as data if default is selected, remove option to have no data selection

mmacke01 committed:
refactored multiPageScreen_addToOptionsList to appendFormElementsToOptions

mmacke01 committed:
updated function call so things aren't broken until my next commit

mmacke01 committed:
fixed dropdown generation in graph cases tab
